fil_fungible_token/src/token/mod.rs

```diff
@@ -113,27 +113,39 @@ where
         initial_owner: &Address,
         amount: &TokenAmount,
         data: &RawBytes,
-    ) -> Result<()> {
+    //) -> Result<()> {
+        ) -> Result<TokenReceivedParams> {
         let amount = validate_amount(amount, "mint", self.granularity)?;
         // init the operator account so that its actor ID can be referenced in the receiver hook
         let operator_id = self.resolve_or_init(operator)?;
         // init the owner account as allowance and balance checks are not performed for minting
         let owner_id = self.resolve_or_init(initial_owner)?;
 
-        let old_state = self.state.clone();
+        // state save/revert will be managed outside of this function
+        //let old_state = self.state.clone();
 
         // Increase the balance of the actor and increase total supply
+        // todo: probably don't need transaction thing here, the entire minting+receiver hook operation belongs in one
```

Comment:
I think it is still a useful property that this token object maintains state that makes sense, even if it returns an error. e.g. caller code could try one thing, then if it fails try another (assuming that no state changes were made, even in memory, by the first op)

Reply:
It didn't seem so useful to me because my original design was going to move the entire mint call into a similar thing. Also wanted to have a "save but revert on failure" version for the receiver hook call, but I think that stuff needs to belong to the state as it's not limited to Token API use (it'd build on my idea from https://github.com/helix-onchain/filecoin/pull/55#discussion_r948616517 ).

I don't think any of this is needed in the abort-on-failure design so I'll remove the comment here and leave the transaction alone.

Reply:
It's a good point for discussion whether we need this transactional behaviour in the state object, or only in the token object. I could easily be convinced of the latter, but let's defer for another PR.